### PR TITLE
Add action to deploy the app to AWS

### PR DESCRIPTION
**Pull request by altonotch**

### CODE REVIEW

File: deployment/Dockerfile.prod [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# base_image is define in peerbook/Dockerfile
+# to build base_image run from peerbook:
+# docker build -t base_image .
+FROM base_image
```
Comment (daonb) [inline deployment/Dockerfile.prod:4]: I don't like having two images especially as one of them can't run. Why not merge this into Dockerfile?

File: deployment/build_and_push.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+set -e
+
+VERSION=$(git rev-parse --short HEAD)
```
Comment (daonb) [inline deployment/build_and_push.sh]: I use SemVer, git tags and a CHANGELOG.md to track version. If you want a version, use the git tag, if you want a HASH call it a HASH.

File: deployment/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+VERSION="$(git rev-parse --short HEAD)"
```
Comment (daonb) [inline deployment/deploy.sh]: HASH or` =git tag -l | tail -1`

File: deployment/docker-compose.template.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+version: '3'
```
Comment (daonb) [inline deployment/docker-compose.template.yml:3]: I'd appreciate a comment at the top line explaining why this file is here

File: deployment/ecs-params.template.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+version: 1
```
Comment (daonb) [inline deployment/ecs-params.template.yml:3]: ditto

**Review by daonb — CHANGES_REQUESTED** [review]
Hi Ana,

Thanks for the clean code.  Maybe I'm missing something but I don't see anyone using `base_image` for anything. Better to keep things simpler and have just one Dockerfile.

File: deployment/Dockerfile.prod (hunk 1 above)
Comment (altonotch) [reply deployment/Dockerfile.prod:4]: Base image is your initial dockerfile, which if not we needed to change in order to have a deployed version. If we merge this into your dockerfile, you and everyone else, will need to start developing differently. The other way to do this, is to utilise docker stages, when developing locally, you will need to remember to build the right stage.

File: deployment/build_and_push.sh (hunk 2 above)
Comment (altonotch) [reply deployment/build_and_push.sh]: This is the unique identifier of an image. Can call it hash, np.

File: deployment/docker-compose.template.yml (hunk 4 above)
Comment (altonotch) [reply deployment/docker-compose.template.yml:3]: This file is the way to deploy to ecs. This whole directory is the deployment scripts required to put the code in aws.

File: deployment/ecs-params.template.yml (hunk 5 above)
Comment (altonotch) [reply deployment/ecs-params.template.yml:3]: Again, you need to know and understand the deployment process to ECS.

File: deployment/docker-compose.template.yml (hunk 4 above)
Comment (altonotch) [reply deployment/docker-compose.template.yml:3]: Added a comment and a link to related aws documentation.

File: deployment/ecs-params.template.yml (hunk 5 above)
Comment (daonb) [reply deployment/ecs-params.template.yml:3]: the average joe just needs to know it's used in the ecs deploy shell script

**Review by daonb — APPROVED** [review]

**Pull request merged**